### PR TITLE
[doc] Update example ES output configs to reflect DS being default

### LINKED ISSUE #14008: Doc: Ingest pipeline info needs an update with respect to data streams
Update config example for data streams. Something like: 

```
input {
  beats {
    port => 5044
  }
}
output {
  if [@metadata][pipeline] {
    elasticsearch {
      hosts => "http://localhost:9200/"
      pipeline => "%{[@metadata][pipeline]}"
      user => "elastic"
      password => "password"
      manage_template => false
      index => "%{[@metadata][beat]}-%{[@metadata][version]}"
      action => "create"
    }
  } else {
    elasticsearch {
      hosts => "http://localhost:9200/"
      user => "elastic"
      password => "password"
      manage_template => false
      index => "%{[@metadata][beat]}-%{[@metadata][version]}"
      action => "create"
    }
  }
}
```
Consider adding data streams config example in addition to current example. Keep existing example available for users not using data streams.

**URL:** 
Example: https://www.elastic.co/guide/en/logstash/current/use-ingest-pipelines.html

### PR DESCRIPTION
**Pull request by roaksoax**
## What does this PR do?

Updates the example configuration in https://www.elastic.co/guide/en/logstash/current/use-ingest-pipelines.html, to reflect the fact that elasticsearch output now defaults to Datastreams.

## Why is it important/What is the impact to the user?

It is important because users using defaults, will incorrectly configure their pipeline by using a 'index' setting that can only be used when datastreams is disabled.

Fixes #14008

### CODE REVIEW

File: docs/static/filebeat-modules.asciidoc [hunk 1]
```diff
@@ -74,23 +74,26 @@ output {
...
       user => "elastic"
       password => "secret"
     }
   }
 }
 -----
-<1> Set the `pipeline` option to `%{[@metadata][pipeline]}`. This setting
+<1> Set the `index` option to `%{[@metadata][beat]}-%{[@metadata][version]}-%{+YYYY.MM.dd}`
+if you are disabling the use of Data Streams in your configuration. Data Streams is
+enabled by default starting from 8.0.
```
Comment (karenzone) [inline docs/static/filebeat-modules.asciidoc:95]:
```suggestion
<1> If you are disabling the Data Streams in your configuration, set the `index` option to `%{[@metadata][beat]}-%{[@metadata][version]}-%{+YYYY.MM.dd}` instead.
Data streams are
enabled by default.
```

Comment (karenzone) [reply docs/static/filebeat-modules.asciidoc:95]: Each minor revision has its own complete doc set (8.2, 8.1, 8.0, 7.17, etc.) that can easily be referenced. For that reason, we don't usually put version numbers in our content.  In the 8.3 version of the guide, saying that data streams are enabled is enough.

File: docs/static/filebeat-modules.asciidoc [hunk 2]
```diff
@@ -74,23 +74,26 @@ output {
...
+<2> Set the `pipeline` option to `%{[@metadata][pipeline]}`. This setting
 configures {ls} to select the correct ingest pipeline based on metadata
 passed in the event.
```
Comment (karenzone) [inline docs/static/filebeat-modules.asciidoc:98]:
```suggestion
<2> Configures {ls} to select the correct ingest pipeline based on metadata
passed in the event.
```

File: docs/static/filebeat-modules.asciidoc (hunk 1 above)
Comment (karenzone) [reply docs/static/filebeat-modules.asciidoc:95]: Elastic standard for data streams is lower case and treated as plural.

**Review by karenzone — COMMENTED** [review]
Nice changes. This approach will help users quickly see the differences with and without data streams. 

I left comments inline for consideration.
Note: I reviewed docs, but I did not validate the Logstash implementation.

**Review by karenzone — CHANGES_REQUESTED** [review]
LGTM

**Review by karenzone — APPROVED** [review]

**Comment by roaksoax** [comment]
@logstashmachine backport 8.4

**Comment by roaksoax** [comment]
@logstashmachine backport 8.3

**Pull request merged**